router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -0,0 +1,178 @@
+}
+
+func (authErrHandler *OAuthErrResHandler) RefreshToken(workspaceId string, accountId string, accessToken string) (statusCode int, respBody string) {
+	authErrHandler.oauthErrHandlerReqTimerStat.Start()
+	if len(strings.TrimSpace(accessToken)) == 0 {
+		return http.StatusBadRequest, `Cannot proceed with refresh token request as accessToken is empty`
+	}
+	authErrHandler.oauthErrHandlerNetReqTimerStat.Start()
+	refreshUrl := fmt.Sprintf("%s/dest/workspaces/%s/accounts/%s/token", configBEURL, workspaceId, accountId)
+	authErrHandler.oauthErrHandlerNetReqTimerStat.End()
+	refTokenBody := RefreshTokenBody{
+		hasExpired:   true,
+		expiredToken: accessToken,
+	}
+	res, err := json.Marshal(refTokenBody)
+	if err != nil {
+		panic(err)
+	}
+	refreshResponse, refreshErr := authErrHandler.client.Post(refreshUrl, "application/json; charset=utf-8", bytes.NewBuffer(res))
+	if refreshErr != nil {
+		return http.StatusBadRequest, err.Error()
+	}
+	statusCode, response := authErrHandler.processResponse(refreshResponse, refreshErr)
+	authErrHandler.oauthErrHandlerReqTimerStat.End()
+	authErrHandler.logger.Debugf("[%s request] :: Refresh token response received : %s", loggerNm, response)
+	return statusCode, response
+}
+
+func (authErrHandler *OAuthErrResHandler) DisableDestination(destination backendconfig.DestinationT, workspaceId string) (statusCode int, respBody string) {
+	authErrHandler.oauthErrHandlerReqTimerStat.Start()
+	destinationId := destination.ID
+	disableURL := fmt.Sprintf("%s/workspaces/%s/destinations/%s/disable", configBEURL, workspaceId, destinationId)
+	req, err := http.NewRequest(http.MethodDelete, disableURL, nil)
+	if err != nil {
+		// Abort on receiving an error in request formation
+		return http.StatusBadRequest, err.Error()
+	}
+	authErrHandler.logger.Debugf("[%s request] :: Disable Request sent : %s", loggerNm)
+	authErrHandler.oauthErrHandlerNetReqTimerStat.Start()
+	res, doErr := authErrHandler.client.Do(req)
+	authErrHandler.oauthErrHandlerNetReqTimerStat.End()
+	if doErr != nil {
+		// Abort on receiving an error
+		return http.StatusBadRequest, err.Error()
+	}
+	statusCode, resp := authErrHandler.processResponse(res, doErr)
+	authErrHandler.oauthErrHandlerReqTimerStat.End()
+	authErrHandler.logger.Debugf("[%s request] :: Disable Response received : %s", loggerNm)
+	return statusCode, resp
+}
+
+func (authErrHandler *OAuthErrResHandler) processResponse(resp *http.Response, err error) (statusCode int, respBody string) {
+	var respData []byte
+	if resp != nil && resp.Body != nil {
+		respData, _ = ioutil.ReadAll(resp.Body)
+		defer resp.Body.Close()
+	}
+	var contentTypeHeader string
+	if resp != nil && resp.Header != nil {
+		contentTypeHeader = resp.Header.Get("Content-Type")
+	}
+	if contentTypeHeader == "" {
+		//Detecting content type of the respBody
+		contentTypeHeader = http.DetectContentType(respData)
```

Comment:
Can't we ignore the part where we fetch the contentType from header and just reply on this?

Reply:
Will change this